cli/azd/pkg/input/console.go

```diff
@@ -17,11 +17,32 @@ import (
 	"github.com/azure/azure-dev/cli/azd/pkg/contracts"
 	"github.com/azure/azure-dev/cli/azd/pkg/output"
 	"github.com/mattn/go-colorable"
+	"github.com/theckman/yacspin"
+)
+
+type MessageUxType int
+
+const (
+	Title MessageUxType = iota
+	Step
+	StepDone
+	StepFailed
+	ResultSuccess
+	ResultError
 )
 
 type Console interface {
 	// Prints out a message to the underlying console write
 	Message(ctx context.Context, message string)
+	// Prints out a message following the UX format type
+	MessageUx(ctx context.Context, message string, format MessageUxType)
+	// Prints progress spinner with the given title.
+	// If a previous spinner is running, the title is updated.
+	ShowSpinner(ctx context.Context, title string, format MessageUxType)
+	// Stop the current spinner from the console and change the spinner bar for the lastMessage
+	// Set lastMessage to empty string to clear the spinner message instead of a displaying a last message
+	// If there is no spinner running, this is a no-op function
+	StopSpinner(ctx context.Context, lastMessage string, format MessageUxType)
```

Comment:
Have you considered introducing an object that manages the progress, I think it reduces responsibility of `Console` and make it easier to use? Like a `Step` or `Operation`

A basic example would look like:
```golang
step := console.DisplayStep("Doing something")
step.Title("Doing something else...")
step.End(Success | Error)
```

Haven't thought through implications / implementation, but the API would look and feel more natural.

Reply:
> an object that manages the progress

That's the Console object already. It handles input and output, and the progress is nothing but an output